library/core/src/panicking.rs

```diff
@@ -59,7 +61,33 @@ pub const fn panic_fmt(fmt: fmt::Arguments<'_>) -> ! {
         fn panic_impl(pi: &PanicInfo<'_>) -> !;
     }
 
-    let pi = PanicInfo::internal_constructor(Some(&fmt), Location::caller(), true);
+    let pi = PanicInfo::internal_constructor(Some(&fmt), Location::caller(), None, true);
+
+    // SAFETY: `panic_impl` is defined in safe Rust code and thus is safe to call.
+    unsafe { panic_impl(&pi) }
+}
+
+#[cold]
+// If panic_immediate_abort, inline the abort call,
+// otherwise avoid inlining because of it is cold path.
+#[cfg_attr(not(feature = "panic_immediate_abort"), inline(never))]
+#[cfg_attr(feature = "panic_immediate_abort", inline)]
+#[track_caller]
+#[rustc_do_not_const_check] // hooked by const-eval
```

Comment:
Since this PR doesn't change anything in `compiler/`, I don't think this function is actually hooked by const-eval.

You either need to make this a lang item and add support in `rustc_const_eval/src/const_eval/machine.rs`, or make this a non-const function.